### DISCUSSION TITLE
[Feature] Memory-friendly search backend instead of typesense

### DISCUSSION THREAD

**Discussion by ghost** [immich-app/immich#4162] [Feature Request]
### The feature

typesense is an in-memory search backend. the embedding data may quickly fill up the memory if one have large number of photos.

Consider [meilisearch](https://www.meilisearch.com/docs/learn/experimental/vector_search) or [qdrant](https://qdrant.tech/).

### Platform

- [x] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
We're very unlikely to move to another typesense-like system, however we are considering getting rid of it entirely and bringing the search back into the main database using pgvector.

**Reply by ghost** [reply]
pgvector is also cool.
